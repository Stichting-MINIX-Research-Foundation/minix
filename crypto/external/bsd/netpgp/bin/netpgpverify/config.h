#ifndef CONFIG_H_
#define CONFIG_H_ 20140308

#ifndef __UNCONST
#define __UNCONST(a)   ((void *)(unsigned long)(const void *)(a))
#endif /* __UNCONST */

#ifndef USE_ARG
#define USE_ARG(x)       /*LINTED*/(void)&(x)
#endif /* USE_ARG */

#endif /* CONFIG_H_ */
