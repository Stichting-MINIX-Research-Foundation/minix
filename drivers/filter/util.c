/* Filter driver - utility functions */

#include "inc.h"
#include <sys/mman.h>
#include <signal.h>

static clock_t next_alarm;

/*===========================================================================*
 *				flt_malloc				     *
 *===========================================================================*/
char *flt_malloc(size_t size, char *sbuf, size_t ssize)
{
	/* Allocate a buffer for 'size' bytes. If 'size' is equal to or less
	 * than 'ssize', return the static buffer 'sbuf', otherwise, use
	 * malloc() to allocate memory dynamically.
	 */
	char *p;

	if (size <= ssize)
		return sbuf;

	if(!(p = alloc_contig(size, 0, NULL)))
		panic("out of memory: %d", size);

	return p;
}

/*===========================================================================*
 *				flt_free				     *
 *===========================================================================*/
void flt_free(char *buf, size_t size, const char *sbuf)
{	
	/* Free a buffer previously allocated with flt_malloc().
	 */

	if(buf != sbuf)
		free_contig(buf, size);
}

/*===========================================================================*
 *				flt_alarm				     *
 *===========================================================================*/
clock_t flt_alarm(clock_t dt)
{
	int r;

	if((int) dt < 0)
		return next_alarm;

	r = sys_setalarm(dt, 0);

	if(r != OK)
		panic("sys_setalarm failed: %d", r);

	if(dt == 0) {
		if(!next_alarm)
			panic("clearing unset alarm: %d", r);
		next_alarm = 0;
	} else {
		if(next_alarm)
			panic("overwriting alarm: %d", r);
		if ((r = getticks(&next_alarm)) != OK)
			panic("getuptime failed: %d", r);
		next_alarm += dt;
	}

	return next_alarm;
}

