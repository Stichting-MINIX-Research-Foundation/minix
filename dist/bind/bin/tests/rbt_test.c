/*
 * Copyright (C) 2004, 2005, 2007, 2009  Internet Systems Consortium, Inc. ("ISC")
 * Copyright (C) 1999-2001  Internet Software Consortium.
 *
 * Permission to use, copy, modify, and/or distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND ISC DISCLAIMS ALL WARRANTIES WITH
 * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
 * AND FITNESS.  IN NO EVENT SHALL ISC BE LIABLE FOR ANY SPECIAL, DIRECT,
 * INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
 * LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE
 * OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
 * PERFORMANCE OF THIS SOFTWARE.
 */

/* $Id: rbt_test.c,v 1.50 2009-09-02 23:48:01 tbox Exp $ */

#include <config.h>

#include <stdlib.h>

#include <isc/commandline.h>
#include <isc/mem.h>
#include <isc/string.h>
#include <isc/util.h>

#include <dns/rbt.h>
#include <dns/fixedname.h>
#include <dns/result.h>

char *progname;
isc_mem_t *mctx;

#define DNSNAMELEN 255

static dns_name_t *
create_name(char *s) {
	int length;
	isc_result_t result;
	isc_buffer_t source, target;
	static dns_name_t *name;

	if (s == NULL || *s == '\0') {
		printf("missing name argument\n");
		return (NULL);
	}

	length = strlen(s);

	isc_buffer_init(&source, s, length);
	isc_buffer_add(&source, length);

	/*
	 * It isn't really necessary in this program to create individual
	 * memory spaces for each name structure and its associated character
	 * string.  It is done here to provide a relatively easy way to test
	 * the callback from dns_rbt_deletename that is supposed to free the
	 * data associated with a node.
	 *
	 * The buffer for the actual name will immediately follow the
	 * name structure.
	 */
	name = isc_mem_get(mctx, sizeof(*name) + DNSNAMELEN);
	if (name == NULL) {
		printf("out of memory!\n");
		return (NULL);
	}

	dns_name_init(name, NULL);
	isc_buffer_init(&target, name + 1, DNSNAMELEN);

	result = dns_name_fromtext(name, &source, dns_rootname, 0, &target);

	if (result != ISC_R_SUCCESS) {
		printf("dns_name_fromtext(%s) failed: %s\n",
		       s, dns_result_totext(result));
		return (NULL);
	}

	return (name);
}

static void
delete_name(void *data, void *arg) {
	dns_name_t *name;

	UNUSED(arg);
	name = data;
	isc_mem_put(mctx, data, sizeof(dns_name_t) + DNSNAMELEN);
}

static void
print_name(dns_name_t *name) {
	isc_buffer_t target;
	char buffer[1024];

	isc_buffer_init(&target, buffer, sizeof(buffer));

	/*
	 * ISC_FALSE means absolute names have the final dot added.
	 */
	dns_name_totext(name, ISC_FALSE, &target);

	printf("%.*s", (int)target.used, (char *)target.base);
}

static void
detail(dns_rbt_t *rbt, dns_name_t *name) {
	dns_name_t *foundname, *origin, *fullname;
	dns_fixedname_t fixedfoundname, fixedorigin, fixedfullname;
	dns_rbtnode_t *node1, *node2;
	dns_rbtnodechain_t chain;
	isc_result_t result;
	isc_boolean_t nodes_should_match = ISC_FALSE;

	dns_rbtnodechain_init(&chain, mctx);

	dns_fixedname_init(&fixedorigin);
	dns_fixedname_init(&fixedfullname);
	dns_fixedname_init(&fixedfoundname);

	origin = dns_fixedname_name(&fixedorigin);
	fullname = dns_fixedname_name(&fixedfullname);
	foundname = dns_fixedname_name(&fixedfoundname);

	node1 = node2 = NULL;

	printf("checking chain information for ");
	print_name(name);
	printf("\n");

	result = dns_rbt_findnode(rbt, name, foundname, &node1, &chain,
				  DNS_RBTFIND_EMPTYDATA, NULL, NULL);

	switch (result) {
	case ISC_R_SUCCESS:
		printf("  found exact.");
		nodes_should_match = ISC_TRUE;
		break;
	case DNS_R_PARTIALMATCH:
		printf("  found parent.");
		break;
	case ISC_R_NOTFOUND:
		printf("  name not found.");
		break;
	default:
		printf("  unexpected result: %s\n", dns_result_totext(result));
		return;
	}

	if (node1 != NULL && node1->data != NULL) {
		printf("  data at node: ");
		print_name(node1->data);
	} else
		printf("  no data at node.");

	if (result == ISC_R_SUCCESS || result == DNS_R_PARTIALMATCH) {
		printf("\n  name from dns_rbt_findnode: ");
		print_name(foundname);
	}

	result = dns_rbtnodechain_current(&chain, foundname, origin, &node2);

	if (result == ISC_R_SUCCESS) {
		printf("\n  name from dns_rbtnodechain_current: ");

		result = dns_name_concatenate(foundname, origin,
					      fullname, NULL);
		if (result == ISC_R_SUCCESS)
			print_name(fullname);
		else
			printf("%s\n", dns_result_totext(result));
		printf("\n      (foundname = ");
		print_name(foundname);
		printf(", origin = ");
		print_name(origin);
		printf(")\n");
		if (nodes_should_match && node1 != node2)
			printf("  nodes returned from each function "
			       "DO NOT match!\n");

	} else
		printf("\n  result from dns_rbtnodechain_current: %s\n",
		       dns_result_totext(result));

	printf("  level_matches = %d, level_count = %d\n",
	       chain.level_matches, chain.level_count);
}

static void
iterate(dns_rbt_t *rbt, isc_boolean_t forward) {
	dns_name_t foundname, *origin;
	dns_rbtnodechain_t chain;
	dns_fixedname_t fixedorigin;
	isc_result_t result;
	isc_result_t (*move)(dns_rbtnodechain_t *chain, dns_name_t *name,
			     dns_name_t *origin);

	dns_rbtnodechain_init(&chain, mctx);

	dns_name_init(&foundname, NULL);
	dns_fixedname_init(&fixedorigin);
	origin = dns_fixedname_name(&fixedorigin);

	if (forward) {
		printf("iterating forward\n" );
		move = dns_rbtnodechain_next;

		result = dns_rbtnodechain_first(&chain, rbt, &foundname,
						origin);

	} else {
		printf("iterating backward\n" );
		move = dns_rbtnodechain_prev;

		result = dns_rbtnodechain_last(&chain, rbt, &foundname,
					       origin);
	}

	if (result != ISC_R_SUCCESS && result != DNS_R_NEWORIGIN)
		printf("start not found!\n");

	else {
		for (;;) {
			if (result == DNS_R_NEWORIGIN) {
				printf("  new origin: ");
				print_name(origin);
				printf("\n");
			}

			if (result == ISC_R_SUCCESS ||
			    result == DNS_R_NEWORIGIN) {
				print_name(&foundname);
				printf("\n");

			} else {
				if (result != ISC_R_NOMORE)
				       printf("UNEXEPCTED ITERATION ERROR: %s",
					      dns_result_totext(result));
				break;
			}

			result = move(&chain, &foundname, origin);
		}
	}
}


#define CMDCHECK(s)	(strncasecmp(command, (s), length) == 0)
#define PRINTERR(r)	if (r != ISC_R_SUCCESS) \
				printf("... %s\n", dns_result_totext(r));

int
main(int argc, char **argv) {
	char *command, *arg, buffer[1024];
	const char *whitespace;
	dns_name_t *name, *foundname;
	dns_fixedname_t fixedname;
	dns_rbt_t *rbt = NULL;
	int length, ch;
	isc_boolean_t show_final_mem = ISC_FALSE;
	isc_result_t result;
	void *data;

	progname = strrchr(*argv, '/');
	if (progname != NULL)
		progname++;
	else
		progname = *argv;

	while ((ch = isc_commandline_parse(argc, argv, "m")) != -1) {
		switch (ch) {
		case 'm':
			show_final_mem = ISC_TRUE;
			break;
		}
	}

	argc -= isc_commandline_index;
	argv += isc_commandline_index;

	if (argc > 1) {
		printf("Usage: %s [-m]\n", progname);
		exit(1);
	}

	setbuf(stdout, NULL);

	/*
	 * So isc_mem_stats() can report any allocation leaks.
	 */
	isc_mem_debugging = ISC_MEM_DEBUGRECORD;

	result = isc_mem_create(0, 0, &mctx);
	if (result != ISC_R_SUCCESS) {
		printf("isc_mem_create: %s: exiting\n",
		       dns_result_totext(result));
		exit(1);
	}

	result = dns_rbt_create(mctx, delete_name, NULL, &rbt);
	if (result != ISC_R_SUCCESS) {
		printf("dns_rbt_create: %s: exiting\n",
		       dns_result_totext(result));
		exit(1);
	}

	whitespace = " \t";

	while (fgets(buffer, sizeof(buffer), stdin) != NULL) {
		length = strlen(buffer);

		if (buffer[length - 1] != '\n') {
			printf("line to long (%lu max), ignored\n",
			       (unsigned long)sizeof(buffer) - 2);
			continue;
		}

		buffer[length - 1] = '\0';

		command = buffer + strspn(buffer, whitespace);

		if (*command == '#')
			continue;

		arg = strpbrk(command, whitespace);
		if (arg != NULL) {
			*arg++ = '\0';
			arg += strspn(arg, whitespace);
		}

		length = strlen(command);
		if (*command != '\0') {
			if (CMDCHECK("add")) {
				name = create_name(arg);
				if (name != NULL) {
					printf("adding name %s\n", arg);
					result = dns_rbt_addname(rbt,
								 name, name);
					PRINTERR(result);
				}

			} else if (CMDCHECK("delete")) {
				name = create_name(arg);
				if (name != NULL) {
					printf("deleting name %s\n", arg);
					result = dns_rbt_deletename(rbt, name,
								    ISC_FALSE);
					PRINTERR(result);
					delete_name(name, NULL);
				}

			} else if (CMDCHECK("nuke")) {
				name = create_name(arg);
				if (name != NULL) {
					printf("nuking name %s "
					       "and its descendants\n", arg);
					result = dns_rbt_deletename(rbt, name,
								    ISC_TRUE);
					PRINTERR(result);
					delete_name(name, NULL);
				}

			} else if (CMDCHECK("search")) {
				name = create_name(arg);
				if (name != NULL) {
					printf("searching for name %s ... ",
					       arg);

					dns_fixedname_init(&fixedname);
					foundname =
						dns_fixedname_name(&fixedname);
					data = NULL;

					result = dns_rbt_findname(rbt, name, 0,
								  foundname,
								  &data);
					switch (result) {
					case ISC_R_SUCCESS:
						printf("found exact: ");
						print_name(data);
						putchar('\n');
						break;
					case DNS_R_PARTIALMATCH:
						printf("found parent: ");
						print_name(data);
						printf("\n\t(foundname: ");
						print_name(foundname);
						printf(")\n");
						break;
					case ISC_R_NOTFOUND:
						printf("NOT FOUND!\n");
						break;
					case ISC_R_NOMEMORY:
						printf("OUT OF MEMORY!\n");
						break;
					default:
						printf("UNEXPECTED RESULT\n");
					}

					delete_name(name, NULL);
				}

			} else if (CMDCHECK("check")) {
				/*
				 * Or "chain".  I know, I know.  Lame name.
				 * I was having a hard time thinking of a
				 * name (especially one that did not have
				 * a conflicting first letter with another
				 * command) that would differentiate this
				 * from the search command.
				 *
				 * But it is just a test program, eh?
				 */
				name = create_name(arg);
				if (name != NULL) {
					detail(rbt, name);

					delete_name(name, NULL);
				}

			} else if (CMDCHECK("forward")) {
				iterate(rbt, ISC_TRUE);

			} else if (CMDCHECK("backward")) {
				iterate(rbt, ISC_FALSE);

			} else if (CMDCHECK("print")) {
				if (arg == NULL || *arg == '\0')
					dns_rbt_printall(rbt);
				else
					printf("usage: print\n");

			} else if (CMDCHECK("quit")) {
				if (arg == NULL || *arg == '\0')
					break;
				else
					printf("usage: quit\n");
			} else {
				printf("a(dd) NAME, d(elete) NAME, "
				       "s(earch) NAME, p(rint), or q(uit)\n");

			}
		}

	}

	dns_rbt_destroy(&rbt);

	if (show_final_mem)
		isc_mem_stats(mctx, stderr);

	return (0);
}
