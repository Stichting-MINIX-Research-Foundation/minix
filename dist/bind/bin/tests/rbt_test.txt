# Copyright (C) 2004, 2007  Internet Systems Consortium, Inc. ("ISC")
# Copyright (C) 1999-2001  Internet Software Consortium.
#
# Permission to use, copy, modify, and/or distribute this software for any
# purpose with or without fee is hereby granted, provided that the above
# copyright notice and this permission notice appear in all copies.
#
# THE SOFTWARE IS PROVIDED "AS IS" AND ISC DISCLAIMS ALL WARRANTIES WITH
# REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
# AND FITNESS.  IN NO EVENT SHALL ISC BE LIABLE FOR ANY SPECIAL, DIRECT,
# INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
# LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE
# OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
# PERFORMANCE OF THIS SOFTWARE.

# $Id: rbt_test.txt,v 1.17 2007-06-19 23:46:59 tbox Exp $

add a.vix.com
add b.vix.com
add c.vix.com
print
add a.b.c.d.e.f.vix.com
add b.b.c.d.e.f.vix.com
add c.b.c.d.e.f.vix.com
print
add a.d.e.f.vix.com
add q.d.e.f.vix.com
add d.e.f.vix.com
print
add g.h.vix.com
print
search q.d.e.f.vix.com
search just-parent.a.vix.com
search no-real-parent.vix.com
search does.not.exist.at.all
forward
backward
# existing name
check vix.com.
# greater than stop node, which has down pointer
check zzz.com.
# less than lowest in level (would be left link from stop node)
check 0.vix.com
# greater than stop node, no down pointer
check d.vix.com
# superdomain stored in existing node
check f.vix.com
# common ancestor stored in existing node; existing is successor
check a.e.f.vix.com
# common ancestor stored in existing node; existing is less but not predecessor
check z.e.f.vix.com
#
check g.vix.com
#
check i.vix.com
#
check b.c.vix.com
nuke d.e.f.vix.com
print
add x.a.vix.com
add y.x.a.vix.com
print
delete a.vix.com
delete x.a.vix.com
print
delete b.vix.com
delete c.vix.com
print
delete y.x.a.vix.com
print
delete g.h.vix.com.
add \[b100000].vix.com.
add \[b010000].vix.com.
add \[b001000].vix.com.
add \[b000100].vix.com.
add \[b000010].vix.com.
add \[b000001].vix.com.
p
search \[b000100].vix.com.
# zap the entire tree
add vix.com.
nuke vix.com.
add a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.
add b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.
print
add .
# zap it again
nuke .
# test splitting of maximal bitstring
add \[xFFFF/16].\[xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/256].com
add \[xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/128].com
print
quit
