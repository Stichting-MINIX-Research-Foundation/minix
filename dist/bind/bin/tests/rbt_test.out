adding name a.vix.com
adding name b.vix.com
adding name c.vix.com
vix.com. (black)
	++ BEG down from vix.com.
	b (black)
		a (RED from b)
			NULL
			NULL
		c (RED from b)
			NULL
			NULL
	-- END down from vix.com.
	NULL
	NULL
adding name a.b.c.d.e.f.vix.com
adding name b.b.c.d.e.f.vix.com
adding name c.b.c.d.e.f.vix.com
vix.com. (black)
	++ BEG down from vix.com.
	b (black)
		a (black from b)
			NULL
			NULL
		c (black from b)
			NULL
			b.c.d.e.f (RED from c)
				++ BEG down from b.c.d.e.f
				b (black)
					a (RED from b)
						NULL
						NULL
					c (RED from b)
						NULL
						NULL
				-- END down from b.c.d.e.f
				NULL
				NULL
	-- END down from vix.com.
	NULL
	NULL
adding name a.d.e.f.vix.com
adding name q.d.e.f.vix.com
adding name d.e.f.vix.com
vix.com. (black)
	++ BEG down from vix.com.
	b (black)
		a (black from b)
			NULL
			NULL
		c (black from b)
			NULL
			d.e.f (RED from c)
				++ BEG down from d.e.f
				b.c (black)
					++ BEG down from b.c
					b (black)
						a (RED from b)
							NULL
							NULL
						c (RED from b)
							NULL
							NULL
					-- END down from b.c
					a (RED from b.c)
						NULL
						NULL
					q (RED from b.c)
						NULL
						NULL
				-- END down from d.e.f
				NULL
				NULL
	-- END down from vix.com.
	NULL
	NULL
adding name g.h.vix.com
vix.com. (black)
	++ BEG down from vix.com.
	b (black)
		a (black from b)
			NULL
			NULL
		d.e.f (black from b)
			++ BEG down from d.e.f
			b.c (black)
				++ BEG down from b.c
				b (black)
					a (RED from b)
						NULL
						NULL
					c (RED from b)
						NULL
						NULL
				-- END down from b.c
				a (RED from b.c)
					NULL
					NULL
				q (RED from b.c)
					NULL
					NULL
			-- END down from d.e.f
			c (RED from d.e.f)
				NULL
				NULL
			g.h (RED from d.e.f)
				NULL
				NULL
	-- END down from vix.com.
	NULL
	NULL
searching for name q.d.e.f.vix.com ... found exact: q.d.e.f.vix.com.
searching for name just-parent.a.vix.com ... found parent: a.vix.com.
	(foundname: a.vix.com.)
searching for name no-real-parent.vix.com ... NOT FOUND!
searching for name does.not.exist.at.all ... NOT FOUND!
iterating forward
  new origin: .
vix.com
  new origin: vix.com.
a
b
c
d.e.f
  new origin: d.e.f.vix.com.
a
b.c
  new origin: b.c.d.e.f.vix.com.
a
b
c
  new origin: d.e.f.vix.com.
q
  new origin: vix.com.
g.h
iterating backward
  new origin: vix.com.
g.h
  new origin: d.e.f.vix.com.
q
  new origin: b.c.d.e.f.vix.com.
c
b
a
  new origin: d.e.f.vix.com.
b.c
a
  new origin: vix.com.
d.e.f
c
b
a
  new origin: .
vix.com
checking chain information for vix.com.
  found exact.  no data at node.
  name from dns_rbt_findnode: vix.com.
  name from dns_rbtnodechain_current: vix.com.
      (foundname = vix.com, origin = .)
  level_matches = 0, level_count = 0
checking chain information for zzz.com.
  name not found.  no data at node.
  name from dns_rbtnodechain_current: g.h.vix.com.
      (foundname = g.h, origin = vix.com.)
  level_matches = 0, level_count = 1
checking chain information for 0.vix.com.
  found parent.  no data at node.
  name from dns_rbt_findnode: vix.com.
  name from dns_rbtnodechain_current: vix.com.
      (foundname = vix.com, origin = .)
  level_matches = 0, level_count = 0
checking chain information for d.vix.com.
  found parent.  no data at node.
  name from dns_rbt_findnode: vix.com.
  name from dns_rbtnodechain_current: c.vix.com.
      (foundname = c, origin = vix.com.)
  level_matches = 0, level_count = 1
checking chain information for f.vix.com.
  found parent.  no data at node.
  name from dns_rbt_findnode: vix.com.
  name from dns_rbtnodechain_current: c.vix.com.
      (foundname = c, origin = vix.com.)
  level_matches = 0, level_count = 1
checking chain information for a.e.f.vix.com.
  found parent.  no data at node.
  name from dns_rbt_findnode: vix.com.
  name from dns_rbtnodechain_current: c.vix.com.
      (foundname = c, origin = vix.com.)
  level_matches = 0, level_count = 1
checking chain information for z.e.f.vix.com.
  found parent.  no data at node.
  name from dns_rbt_findnode: vix.com.
  name from dns_rbtnodechain_current: q.d.e.f.vix.com.
      (foundname = q, origin = d.e.f.vix.com.)
  level_matches = 0, level_count = 2
checking chain information for g.vix.com.
  found parent.  no data at node.
  name from dns_rbt_findnode: vix.com.
  name from dns_rbtnodechain_current: q.d.e.f.vix.com.
      (foundname = q, origin = d.e.f.vix.com.)
  level_matches = 0, level_count = 2
checking chain information for i.vix.com.
  found parent.  no data at node.
  name from dns_rbt_findnode: vix.com.
  name from dns_rbtnodechain_current: g.h.vix.com.
      (foundname = g.h, origin = vix.com.)
  level_matches = 0, level_count = 1
checking chain information for b.c.vix.com.
  found parent.  data at node: c.vix.com.
  name from dns_rbt_findnode: c.vix.com.
  name from dns_rbtnodechain_current: c.vix.com.
      (foundname = c, origin = vix.com.)
  level_matches = 1, level_count = 1
nuking name d.e.f.vix.com and its descendants
vix.com. (black)
	++ BEG down from vix.com.
	b (black)
		a (black from b)
			NULL
			NULL
		g.h (black from b)
			c (RED from g.h)
				NULL
				NULL
			NULL
	-- END down from vix.com.
	NULL
	NULL
adding name x.a.vix.com
adding name y.x.a.vix.com
vix.com. (black)
	++ BEG down from vix.com.
	b (black)
		a (black from b)
			++ BEG down from a
			x (black)
				++ BEG down from x
				y (black)
					NULL
					NULL
				-- END down from x
				NULL
				NULL
			-- END down from a
			NULL
			NULL
		g.h (black from b)
			c (RED from g.h)
				NULL
				NULL
			NULL
	-- END down from vix.com.
	NULL
	NULL
deleting name a.vix.com
deleting name x.a.vix.com
vix.com. (black)
	++ BEG down from vix.com.
	b (black)
		a (black from b)
			++ BEG down from a
			x (black)
				++ BEG down from x
				y (black)
					NULL
					NULL
				-- END down from x
				NULL
				NULL
			-- END down from a
			NULL
			NULL
		g.h (black from b)
			c (RED from g.h)
				NULL
				NULL
			NULL
	-- END down from vix.com.
	NULL
	NULL
deleting name b.vix.com
deleting name c.vix.com
vix.com. (black)
	++ BEG down from vix.com.
	g.h (black)
		a (RED from g.h)
			++ BEG down from a
			x (black)
				++ BEG down from x
				y (black)
					NULL
					NULL
				-- END down from x
				NULL
				NULL
			-- END down from a
			NULL
			NULL
		NULL
	-- END down from vix.com.
	NULL
	NULL
deleting name y.x.a.vix.com
vix.com. (black)
	++ BEG down from vix.com.
	g.h (black)
		a (RED from g.h)
			++ BEG down from a
			x (black)
				NULL
				NULL
			-- END down from a
			NULL
			NULL
		NULL
	-- END down from vix.com.
	NULL
	NULL
deleting name g.h.vix.com.
adding name \[b100000].vix.com.
adding name \[b010000].vix.com.
adding name \[b001000].vix.com.
adding name \[b000100].vix.com.
adding name \[b000010].vix.com.
adding name \[b000001].vix.com.
vix.com. (black)
	++ BEG down from vix.com.
	\[x80/6] (black)
		\[x0/1] (RED from \[x80/6])
			++ BEG down from \[x0/1]
			\[x80/5] (black)
				\[x0/1] (RED from \[x80/5])
					++ BEG down from \[x0/1]
					\[x8/4] (black)
						\[x0/1] (RED from \[x8/4])
							++ BEG down from \[x0/1]
							\[x8/3] (black)
								\[x0/1] (RED from \[x8/3])
									++ BEG down from \[x0/1]
									\[x8/2] (black)
										\[x4/2] (RED from \[x8/2])
											NULL
											NULL
										NULL
									-- END down from \[x0/1]
									NULL
									NULL
								NULL
							-- END down from \[x0/1]
							NULL
							NULL
						NULL
					-- END down from \[x0/1]
					NULL
					NULL
				NULL
			-- END down from \[x0/1]
			NULL
			NULL
		a (RED from \[x80/6])
			++ BEG down from a
			x (black)
				NULL
				NULL
			-- END down from a
			NULL
			NULL
	-- END down from vix.com.
	NULL
	NULL
searching for name \[b000100].vix.com. ... found exact: \[x10/6].vix.com.
adding name vix.com.
nuking name vix.com. and its descendants
adding name a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.
adding name b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.
b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w. (black)
	++ BEG down from b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.
	a (black)
		NULL
		NULL
	-- END down from b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.x.y.z.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p.q.r.s.t.u.v.w.
	NULL
	NULL
adding name .
nuking name . and its descendants
adding name \[xFFFF/16].\[xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/256].com
adding name \[xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/128].com
\[xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/128].com. (black)
	++ BEG down from \[xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/128].com.
	\[xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/144] (black)
		NULL
		NULL
	-- END down from \[xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/128].com.
	NULL
	NULL
